fields/types/relationship/RelationshipType.js

```diff
@@ -244,7 +241,7 @@ relationship.prototype.updateItem = function (item, data, callback) {
 		if (value && value !== item.get(this.path)) {
 			// If it's set and has changed, I do.
 			item.set(this.path, value);
-		} else if (!value && item.get(this.path)) {
+		} else if ((value === undefined || !value) && item.get(this.path))  {
```

Comment:
the check for !value will pick up on the `value === undefined`, so this can be simplified

```suggestion
		} else if (!value && item.get(this.path))  {
```